src/app/src/test/kotlin/org/imperial/mrc/hint/integration/ADRTests.kt

```diff
@@ -56,6 +56,31 @@ class ADRTests : SecureIntegrationTests() {
         }
     }
 
+    @ParameterizedTest
+    @EnumSource(IsAuthorized::class)
+    fun `can get ADR datasets`(isAuthorized: IsAuthorized) {
+        testRestTemplate.postForEntity<String>("/adr/key", getPostEntity())
+        val result = testRestTemplate.getForEntity<String>("/adr/datasets")
+
+        assertSecureWithSuccess(isAuthorized, result, null)
+
+        if (isAuthorized == IsAuthorized.TRUE) {
+            val data = ObjectMapper().readTree(result.body!!)["data"]
+            assertThat(data.isArray).isTrue()
+            // we used a fake key so no results will come back
+            assertThat(data.count()).isEqualTo(0)
+        }
+
+        val resultWithResources = testRestTemplate.getForEntity<String>("/adr/datasets?showInaccessible=true")
+        if (isAuthorized == IsAuthorized.TRUE) {
+            val data = ObjectMapper().readTree(resultWithResources.body!!)["data"]
+            assertThat(data.isArray).isTrue()
+            // with the showInaccessible flag set, we can see packages
+            // that aren't actually accessible to our fake user
+            assertThat(data.count()).isEqualTo(3)
```

Comment:
So these are all the datasets in the unaids schema? Can we rely on that to stay as 3, or might some new stuff get added causing this test to fail?

Reply:
On the ADR dev server. But in fact it's now 1 lol, I'll change this to > 0.